### ISSUE TITLE
Domain Picker: check availability for the selected domain

### ISSUE THREAD

**Issue by razvanpapadopol** [Automattic/wp-calypso#48634]
**tl;dr**: The check we've implemented in #45698 for domain step in Gutenboarding should be done inside `DomainPicker` component whenever `currentDomain` prop changes.

**Why**
- The user needs to be informed if the selected domain is unavailable ASAP.

**What**
- We need to check if the selected domain is available in all scenarios, including when it's coming from a cart item or local storage (previous selection).
- We need to do this check in Gutenboarding and launch flows (and the future Post-publish domains upsell) so the best way to do this is inside `DomainPicker` package.

**AC**

- [ ] Do the availability check for each selected domain and render _Unavailable_ status depending on the API response.
- [ ] In Gutenboarding, since the availability check is done as soon as a domain is selected, advancing from the domain step should happen immediately.
